app/(main)/league/all/page.test.tsx

```diff
@@ -1,124 +1,235 @@
+        participants: ['123456', '78'],
+        survivors: ['123456', '78', '9'],
+      },
+    ],
+  })),
 }));
 
 jest.mock('@/utils/utils', () => ({
   getUserLeagues: jest.fn(() => Promise.resolve([])),
 }));
 
 jest.mock('@/api/apiFunctions', () => ({
-  getGameWeek: jest.fn(() =>
-    Promise.resolve({
-      week: 1,
-    }),
-  ),
   getAllLeagues: jest.fn(() =>
     Promise.resolve([
       {
         leagueId: '123',
         leagueName: 'Test League',
-        logo: 'https://example.com/logo.png',
+        logo: 'logo.png',
         participants: ['123456', '78', '9'],
         survivors: ['123456', '78'],
       },
     ]),
   ),
+  addUserToLeague: jest.fn(() => Promise.resolve({})),
 }));
 
-jest.mock('@/context/AuthContextProvider', () => ({
-  useAuthContext: jest.fn(() => ({ user: { id: '123' } })),
+jest.mock('react-hot-toast', () => ({
+  toast: {
+    custom: jest.fn(),
+  },
 }));
 
 describe('Leagues Component', () => {
   const mockUseDataStore = useDataStore as unknown as jest.Mock;
   const mockGetUserLeagues = getUserLeagues as jest.Mock;
-  const mockGetGameWeek = getGameWeek as jest.Mock;
   const mockGetAllLeagues = getAllLeagues as jest.Mock;
+  const mockAddUserToLeague = addUserToLeague as jest.Mock;
 
   beforeEach(() => {
-    jest.clearAllMocks();
+    jest.resetAllMocks();
+    mockUseAuthContext.isSignedIn = false;
   });
 
-  xtest('should render "You are not enrolled in any leagues" message when no leagues are found', async () => {
-    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
+  test('should render "You are not enrolled in any leagues" message when no leagues are found', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    mockUseDataStore.mockReturnValue({
+      user: { id: '123', leagues: [] },
+      allLeagues: [],
+    });
     mockGetUserLeagues.mockResolvedValueOnce([]);
-    mockGetGameWeek.mockResolvedValueOnce({ week: 1 });
+
     render(<Leagues />);
 
-    await waitFor(() => {
-      expect(
-        screen.getByText('You are not enrolled in any leagues'),
-      ).toBeInTheDocument();
+    await waitForElementToBeRemoved(() => screen.getByTestId('global-spinner'));
+
+    expect(screen.getByTestId('no-leagues-message')).toBeInTheDocument();
+  });
+
+  test('should display GlobalSpinner while loading data', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    mockUseDataStore.mockReturnValueOnce({
+      user: { id: '123', leagues: [] },
+      allLeagues: [],
     });
+
+    render(<Leagues />);
+
+    expect(screen.getByTestId('global-spinner')).toBeInTheDocument();
   });
 
-  xtest('should display GlobalSpinner while loading data', async () => {
-    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
+  test('should not display GlobalSpinner after loading data', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    mockUseDataStore.mockReturnValue({
+      user: { id: '123', leagues: [] },
```

Comment:
we need to make sure we are mocking the correct data models. documentId and email is missing from user.